### PR TITLE
default_value_column_iterator

### PR DESCRIPTION
**Pull request by zuochunwei**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/column_reader.cpp [hunk 1]
```diff
@@ -760,5 +761,55 @@ Status DefaultValueColumnIterator::next_batch(size_t* n, ColumnBlockView* dst, b
...
+        assert(_type_size == sizeof(FieldTypeTraits<OLAP_FIELD_TYPE_DECIMAL>::CppType)); //decimal12_t
+        decimal12_t* d = (decimal12_t*)_mem_value;
+        DecimalV2Value data(d->integer, d->fraction);
+        vectorized::Int128 &x = data.value();
+        return std::string((char*)&x, sizeof(x));
+    } else {
+        return std::string((char*)_mem_value, _type_size);
+    }
+
+    return std::string();
```
Comment (HappenLee) [inline be/src/olap/rowset/segment_v2/column_reader.cpp:796]: 这个函数是不可能走到的把，可以删除

File: be/src/olap/rowset/segment_v2/column_reader.cpp [hunk 2]
```diff
@@ -760,5 +761,55 @@ Status DefaultValueColumnIterator::next_batch(size_t* n, ColumnBlockView* dst, b
...
+        vectorized::Int128 x = binary_cast<DateTimeValue, vectorized::Int128>(value);
+        return std::string((char*)&x, sizeof(x));
+    } else if (type == OLAP_FIELD_TYPE_DECIMAL) {
+        assert(_type_size == sizeof(FieldTypeTraits<OLAP_FIELD_TYPE_DECIMAL>::CppType)); //decimal12_t
+        decimal12_t* d = (decimal12_t*)_mem_value;
+        DecimalV2Value data(d->integer, d->fraction);
+        vectorized::Int128 &x = data.value();
+        return std::string((char*)&x, sizeof(x));
+    } else {
+        return std::string((char*)_mem_value, _type_size);
```
Comment (HappenLee) [inline be/src/olap/rowset/segment_v2/column_reader.cpp:793]: 这里会导致一次内存分配的问题，得想办法能不能规避一下。或者把column直接作为参数插入，不要返回std::string了

File: be/src/olap/rowset/segment_v2/column_reader.cpp (hunk 1 above)
Comment (zuochunwei) [reply be/src/olap/rowset/segment_v2/column_reader.cpp:796]: ok

File: be/test/olap/rowset/segment_v2/column_reader_writer_test.cpp [hunk 3]
```diff
@@ -516,6 +519,95 @@ void test_read_default_value(string value, void* result) {
...
+            vectorized::MutableColumnPtr mcp = create_vectorized_column_ptr(type);
+
+            size_t rows_read = 16;
+            bool has_null;
+            st = iter.next_batch(&rows_read, mcp, &has_null);
+
+            ASSERT_TRUE(st.ok());
+            for (int j = 0; j < rows_read; ++j) {
+                if (type == OLAP_FIELD_TYPE_CHAR) {
+                    //ASSERT_EQ(*(string*)result, reinterpret_cast<const Slice*>(col.cell_ptr(j))->to_string()) << "j:" << j;
```
Comment (HappenLee) [inline be/test/olap/rowset/segment_v2/column_reader_writer_test.cpp:576]: 还有这里

**Pull request merged**